### DISCUSSION TITLE
HTTPS support loading OpenSSL dynamically

### DISCUSSION THREAD

**Discussion by dacap** [curl/curl#8224] [Ideas]
Hi there 👋  I'm not sure if this is the correct place to bring this idea up (or if it was talked in the past).

Do you know if it would be possible the create a new TLS backend using the same code of the OpenSSL one but loading the openssl.so library dynamically?

I'm interested on using Schannel on Windows, Secure transport on macOS, and OpenSSL on Linux but without creating a dependency with the openssl.so library i.e. if it doesn't exist, the program should be able to continue running anyway when it's launched.

**Comment by gvanem** [comment]
Qt and CopperSpice does this when `QT_LINKED_OPENSSL` is not defined.
E.g.  `SSL *ssl = q_SSL_new(ctx);`  uses OpenSSL dynamically via some macro. Ref. `DEFINEFUNC()` in:
https://github.com/copperspice/copperspice/blob/master/src/network/ssl/qsslsocket_openssl_symbols_p.h

**Comment by bagder** [comment]
Sure that is possible. I would however advice against it and would object to merging support for it.

1. Everything is going HTTPS these days, curl without TLS capabilities is crippled
2. Dynamically loading and using shared libraries is a road riddled with issues and problems (SONAME is one just one of them), many of which becomes very hard for users to understand and solve
3. Added complexity for very little benefit
